native/src/main/java/io/ballerina/stdlib/nats/basic/client/Request.java

```diff
@@ -71,38 +71,41 @@ public static Object requestMessage(Environment environment, BObject clientObj,
-                int valueInMilliSeconds = (valueInSeconds).multiply(MILLISECOND_MULTIPLIER).intValue();
-                reply = incoming.get(valueInMilliSeconds, TimeUnit.MILLISECONDS);
-            } else {
-                reply = incoming.get();
-            }
-            RecordType recordType = Utils.getRecordType(bTypedesc);
+        return environment.yieldAndRun(() -> {
+            try {
+                Message reply;
+                Future<Message> incoming = natsConnection.request(subject, byteContent);
+                natsMetricsReporter.reportRequest(subject, byteContent.length);
+                if (TypeUtils.getType(duration).getTag() == TypeTags.DECIMAL_TAG) {
+                    BigDecimal valueInSeconds = ((BDecimal) duration).decimalValue();
+                    int valueInMilliSeconds = (valueInSeconds).multiply(MILLISECOND_MULTIPLIER).intValue();
+                    reply = incoming.get(valueInMilliSeconds, TimeUnit.MILLISECONDS);
+                } else {
+                    reply = incoming.get();
+                }
+                RecordType recordType = Utils.getRecordType(bTypedesc);
 
-            BMap<BString, Object> msgRecord = ValueCreator.createRecordValue(recordType);
-            Map<String, Field> fieldMap = recordType.getFields();
-            Type contentType = getReferredType(fieldMap.get(Constants.MESSAGE_CONTENT).getFieldType());
+                BMap<BString, Object> msgRecord = ValueCreator.createRecordValue(recordType);
+                Map<String, Field> fieldMap = recordType.getFields();
+                Type contentType = getReferredType(fieldMap.get(Constants.MESSAGE_CONTENT).getFieldType());
 
-            BMap<BString, Object> populatedRecord = ValueCreator.createRecordValue(msgRecord,
-                    Utils.getValueWithIntendedType(contentType, reply.getData()),
-                    StringUtils.fromString(reply.getSubject()),
-                    StringUtils.fromString(reply.getReplyTo()));
-            boolean constraintValidation = (boolean) clientObj.getNativeData(CONSTRAINT_VALIDATION);
-            validateConstraints(populatedRecord, getElementTypeDescFromArrayTypeDesc(bTypedesc), constraintValidation);
-            natsMetricsReporter.reportResponse(subject);
-            return populatedRecord;
-        } catch (IllegalArgumentException | IllegalStateException | ExecutionException | InterruptedException |
-                TimeoutException ex) {
-            natsMetricsReporter.reportProducerError(subject, NatsObservabilityConstants.ERROR_TYPE_REQUEST);
-            return Utils.createNatsError("Error while requesting message to subject " + subject + ".", ex);
-        } catch (BError bError) {
-            natsMetricsReporter.reportProducerError(subject, NatsObservabilityConstants.ERROR_TYPE_REQUEST);
-            return bError;
-        }
+                BMap<BString, Object> populatedRecord = ValueCreator.createRecordValue(msgRecord,
```

Comment:
This should not be inside yieldAndRun since it will call ballerina runtime

Reply:
Changed in b692a4439d04015786e76ad2e53be8554425181c. Will this work?